test/Microsoft.VisualStudio.Composition.Analyzers.Tests/VSMEF006ImportNullabilityAnalyzerTests.cs

```diff
@@ -0,0 +1,421 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.VisualStudio.Composition.Analyzers;
+using VerifyCS = CSharpCodeFixVerifier<Microsoft.VisualStudio.Composition.Analyzers.VSMEF006ImportNullabilityAnalyzer, Microsoft.VisualStudio.Composition.Analyzers.CodeFixes.VSMEF006ImportNullabilityCodeFixProvider>;
+
+public class VSMEF006ImportNullabilityAnalyzerTests
+{
+    [Fact]
+    public async Task ImportWithoutNullableAnnotations_NoWarning()
+    {
+        string test = """
+            using System.ComponentModel.Composition;
+
+            class Foo
+            {
+                [Import]
+                public string Value { get; set; }
+            }
+            """;
+
+        await VerifyCS.VerifyAnalyzerAsync(test);
+    }
+
+    [Fact]
+    public async Task NullableImportWithAllowDefault_NoWarning()
+    {
+        string test = """
+            #nullable enable
+            using System.ComponentModel.Composition;
+
+            class Foo
+            {
+                [Import(AllowDefault = true)]
+                public string? Value { get; set; }
+            }
+            """;
+
+        await VerifyCS.VerifyAnalyzerAsync(test);
+    }
+
+    [Fact]
+    public async Task NonNullableImportWithoutAllowDefault_NoWarning()
+    {
+        string test = """
+            #nullable enable
+            using System.ComponentModel.Composition;
+
+            class Foo
+            {
+                [Import]
+                public string Value { get; set; }
+            }
+            """;
+
+        await VerifyCS.VerifyAnalyzerAsync(test);
+    }
+
+    [Fact]
+    public async Task NullableImportWithoutAllowDefault_Warning()
+    {
+        string test = """
+            #nullable enable
+            using System.ComponentModel.Composition;
+
+            class Foo
+            {
+                [Import]
+                public string? {|#0:Value|} { get; set; }
```

Comment:
The trouble with this is removing the nullable annotation produces a C# warning because the (implicit) constructor doesn't guarantee the property is initialized with a non-null value.
This _can_ be addressed using either of these syntaxes:

```cs
[Import]
public required string Value { get; set; }
```

or 

```cs
[Import]
public string Value { get; set; } = null!;
```

I prefer the first option, though it requires a polyfill for .NET Framework/Standard folks.
The issue with this diagnostic would be that people won't know how to fix it, unless the diagnostic message volunteers the available workarounds.

Reply:
It's easier to append ` = null!` for these, so I'll go that route for now. Understanding when `required` is available is complex, especially for in-proc VS components that are running on framework. I am not sure we could easily bake in knowledge of all the polyfills that people use. If the goal is to have the codefix produce immediately-compilable code, then `= null!` is safest.